s100py/s102.py

```diff
@@ -1245,101 +1245,165 @@ def __init__(self, *args, **kywrds):
-# # -36.03
-# numpy.max(v["S102_Elevation"])
-# # 1000000.0
-# depths = v["S102_Elevation"]
-# numpy.set_printoptions(precision=2, suppress=True, linewidth=200)
-# print(depths[::300, ::300])
-# new_sample= r"C:\downloads\S102\S102_Update__from_NAVO_for_edition_2.0.0_June2019\BAG_to_S102_converter_v2_0\BAG_to_S102_converter\sample_data\102NOAA_LA_LB_AREA_GEO_%d.h5"
-# h5py.File(new_sample, "r", driver="family", memb_size=681574400)
-# # <HDF5 file "102NOAA_LA_LB_AREA_GEO_%d.h5" (mode r)>
-# nf = h5py.File(new_sample, "r", driver="family", memb_size=681574400)
-# list(nf.attrs.items())
-# # [('eastBoundLongitude', -118.182045), ('epoch', b'20131016'), ('geographicIdentifier', b'Long Beach, CA'), ('horizontalDatumReference', b'EPSG'), ('horizontalDatumValue', 4326), ('issueDate', b'2018/6/28'), ('metaFeatures', b'sample_data/102NOAA_LA_LB_AREA_GEO.gml'), ('metadata', b'sample_data/102NOAA_LA_LB_AREA_GEO.xml'), ('northBoundLatitude', 33.92136), ('productSpecification', b'BAG'), ('southBoundLatitude', 33.780685), ('timeOfIssue', b'2018/6/28'), ('westBoundLongitude', -118.29966)]
-# list(nf.keys())
-# # ['BathymetryCoverage', 'Group_F', 'TracklingListCoverage']
-# list(f.keys())
-# # ['Group_F', 'S102_Grid', 'S102_Tracking_List', 'boundingBox', 'sequenceRule']
-# nd = nf['BathymetryCoverage']
-# list(nd.keys())
-# # ['BathymetryCoverage_01', 'axisNames']
-# nd01 = nd['BathymetryCoverage_01']
-# list(nd01.keys())
-# # ['Group_001']
-# ng = nd01['Group_001']
-# list(ng.keys())
-# # ['values']
-# nv = ng["values"]
-# nv.shape
-# # (3111, 2601)
-# type(nv)
-# # <class 'h5py._hl.dataset.Dataset'>
-# nv.dtype
-# # dtype([('depth', '<f4'), ('uncertainty', '<f4')])
-# nv.compression
-# # 'gzip'
-# v.compression
-# nv.fillvalue
-# # (0., 0.)
-# v.fillvalue
-# # (0., 0.)
-# ndepths=nv["depth"]
-# print(ndepths[::300, ::300])
-# debug = r"C:\Git_Repos\BagToS102\x64\Debug\test2_0_debug_%d.h5"
-# nf = h5py.File(debug, "r", driver="family", memb_size=681574400)
-# list(nf.attrs.items())
-# # [('eastBoundLongitude', 491710.0), ('epoch', b'20131016'), ('geographicIdentifier', b'Long Beach, CA'), ('horizontalDatumReference', b'EPSG'), ('horizontalDatumValue', 4326), ('issueDate', b'2019-02-05'), ('metaFeatures', b'.\\test2_0_debug.gml'), ('metadata', b'.\\test2_0_debug.xml'), ('northBoundLatitude', 5719550.0), ('productSpecification', b'BAG'), ('southBoundLatitude', 5691370.0), ('timeOfIssue', b'2019-02-05'), ('westBoundLongitude', 458060.0)]
-# list(nf.keys())
-# # ['BathymetryCoverage', 'Group_F', 'TracklingListCoverage']
-# nd = nf['BathymetryCoverage']
-# list(nd.keys())
-# # ['BathymetryCoverage_01', 'axisNames']
-# nd01 = nd['BathymetryCoverage_01']
-# ng = nd01['Group_001']
-# nv = ng["values"]
-# nv.dtype
-# # dtype([('depth', '<f4'), ('uncertainty', '<f4')])
-# nv.compression
-# ndepths=nv["depth"]
-# ndepths.size
-# # 9482570
-# ndepths.shape
-# # (2818, 3365)
-# print(ndepths[::300, ::300])
+def make_s102(output_path, elev_raster, uncert_raster, metadata):
```

Comment:
This function should be back in the bag_to_s102 file and not in s102.py.  My reason is that I'm envisioning "csar_to_s102" or geotif_to_s102 and the metadata dict is not defined in S102.py and at this point is Bag specific, so it should be in the bag_to_s102.  If there was a defined dictionary structure that would convert to s102 then this would make sense here.  

I see why you would want a 'make_s102' easy button though.  What would make sense to me is for a make_s102_from_raster that returns a S102File object and fills the raster info (i.e. leave out the metadata part).  That would do some of the setup and then leave it to the utility code, like bag_to_s102 to fill the remaining metadata.  A s102_from_gdal may make sense too, meaning that it could grab raster data and projection (leaving uncertainty empty?).